### ISSUE TITLE
[Bug]: Layout rules (e.g. LT01) flag violations inside template-generated regions where SQLFluff suppresses them

### ISSUE THREAD

**Issue by ryan-pip** [quarylabs/sqruff#2607]
## Version

sqruff 0.38.0 (pipx install), macOS arm64

## Description

When jinja loops/macros generate SQL output, sqruff applies layout rules (LT01 spacing, etc.) to that compiled output. SQLFluff treats template-generated regions as non-fixable and suppresses layout violations within them. This causes sqruff to surface "false positive" lint errors that users can't fix (the source is a `{% for %}` loop, not raw SQL).

This appears related to the parity-gap meta #2498 but isn't explicitly listed there.

## Minimal repro

`.sqlfluff`:
```ini
[sqlfluff]
templater = jinja
dialect = ansi

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = trailing
```

`repro.sql`:
```sql
{% set fields = ['a', 'b', 'c'] %}
select *
from t
where col in (
    {%- for f in fields -%}
        '{{ f }}'{% if not loop.last %},{% endif %}
    {%- endfor -%}
)
```

Run both:
- `sqruff lint repro.sql`
- `sqlfluff lint repro.sql`

## Expected

Both tools clean — the comma-separated list is generated by the for-loop, the user can't insert spaces between its elements without changing the loop body.

## Actual

`sqlfluff lint repro.sql` → `All Finished!` (clean)

`sqruff lint repro.sql`:
```
L:   2 | P:   1 | LT01 | Expected single whitespace between "," and "'b'". [layout.spacing]
L:   2 | P:   5 | LT01 | Expected single whitespace between "," and "'c'". [layout.spacing]
```

**Comment by mikills** [comment]
Thank you for raising this! This issue has now been fixed by #2824. Verified the repro now lints cleanly with the Jinja templater, so closing as fixed.
